### PR TITLE
Add Upgradeable-to annotation for 4.14 GCP cluster

### PR DESCRIPTION
**Pull request by jianping-shu**
The background is addressed in [OCPQE-19413](https://issues.redhat.com/browse/OCPQE-19413), the upgradeable-to annotation is required in the case unless "--force" is added for the upgrade.

### CODE REVIEW

**Comment by jianping-shu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-4.15-upgrade-from-stable-4.14-gcp-ipi-ovn-ipsec-f14

**Comment by jianping-shu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-4.15-upgrade-from-stable-4.14-gcp-ipi-ovn-ipsec-f14

**Comment by jianping-shu** [comment]
@jianlinliu @shellyyang1989 PTAL, thanks

File: ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh [hunk 1]
```diff
@@ -136,15 +136,26 @@ function update_cloud_credentials_oidc(){
...
     fi
 
     local cco_mode; cco_mode="$(oc get cloudcredential cluster -o jsonpath='{.spec.credentialsMode}')"
-    if [[ ${cco_mode} != "Manual" ]]; then
-        echo "CCO annotation change is not required in non-manual mode" && return
+    local platform; platform="$(oc get infrastructure cluster -o jsonpath='{.status.platformStatus.type}')"
+    if [[ ${cco_mode} == "Manual" ]]; then
+        echo "CCO annotation change is required in Manual mode"
+    elif [[ -z "${cco_mode}" || ${cco_mode} == "Mint" ]]; then
+        if [[ "${SOURCE_MINOR_VERSION}" == "14" && ${platform} == "GCP" ]] ; then
```
Comment (shellyyang1989) [inline ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh:150]: Does 4.14 to 4.14 upgrade need the annotation change?

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh:150]: I guess no, the z-stream upgrade would be skipped on line 141.

Comment (jianping-shu) [reply ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh:150]: No. This upgradeable annotation is not needed for z-stream.

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-4.15-upgrade-from-stable-4.14-gcp-ipi-proxy-private-f14

**Pull request merged**